rustls/src/error.rs

```diff
@@ -107,6 +110,24 @@ pub enum Error {
     Other(OtherError),
 }
 
+/// Specific failure cases from [`crate::crypto::signer::CertifiedKey::keys_match`].
+#[non_exhaustive]
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum InconsistentKeys {
+    /// The public key returned by the [`crate::crypto::signer::SigningKey`] does not match that in the certificate.
+    KeyMismatch,
```

Comment:
minor editorial nit: I would be tempted to doc this a little bit more specifically:

```suggestion
    /// The public key returned by the [`crate::crypto::signer::SigningKey`] does not match the public key information in the certificate.
    KeyMismatch,
```